### DISCUSSION TITLE
Project restart: a lot of improvements!

### DISCUSSION THREAD

**Discussion by kkovaletp** [photoview/photoview#973] [Show and tell]
## Summary
This is the 1st release after project restart. It contains a lot of fixes, improvements, and even refactoring. We've included the most important changes, contributed by the Photoview Community while the project was on hold and during these few months after the restart.

We hope that future releases won't take so much time and will contain smaller lists of changes ;-)

### Added
* Add support for .CR3 photos by @PJ-Watson in https://github.com/photoview/photoview/pull/728
* Add swipe gesture by @eadisor in https://github.com/photoview/photoview/pull/651
* Add globe view of Mapbox in Places page by @noel123007 in https://github.com/photoview/photoview/pull/774
* Add the Turkish language by @queeup in https://github.com/photoview/photoview/pull/780
* Add support to parse DateTimes with UTC+Offset by @yosev in https://github.com/photoview/photoview/pull/802
* Add Ukrainian translation by @kkovaletp in https://github.com/photoview/photoview/pull/875
* Add cross-compile support for arm64 to amd64 by @cicadabear in https://github.com/photoview/photoview/pull/888
* (docker-compose): add SQLite variable and volumes example by @alcaprar in https://github.com/photoview/photoview/pull/851

### Improved
* Thumbnail Rendering Options by @PJ-Watson in https://github.com/photoview/photoview/pull/725
* Config🔧:  Improved Simplified Chinese language translation by @rainpenber in https://github.com/photoview/photoview/pull/749
* Restyle titles and spacing in the timeline gallery by @orellazri in https://github.com/photoview/photoview/pull/752
* Limit thumbnail size of the small original photo in the new branch by @WindLi001 in https://github.com/photoview/photoview/pull/798
* Updated Translation for German by @secco04 in https://github.com/photoview/photoview/pull/841
* Update German translation by @secco04 in https://github.com/photoview/photoview/pull/842
* Update Italian translation by @davidecavestro in https://github.com/photoview/photoview/pull/873
* Reordered CLI arguments for FFmpeg to improve speed to generate thumbnails of videos by @AndGem in https://github.com/photoview/photoview/pull/917
* Update Polish translation by @DorianMazur in https://github.com/photoview/photoview/pull/909
* Faster start of video playback by @kabakaev in https://github.com/photoview/photoview/pull/883
* Don't stop scanning the album on media fail by @kkovaletp in https://github.com/photoview/photoview/pull/892
* Remove login page non-empty password requirement by @subtlepseudonym in https://github.com/photoview/photoview/pull/828
* Make face detection optional at build-time by @emersion in https://github.com/photoview/photoview/pull/881
* Remove logging of database URL in place for just dialect by @jordy2254 in https://github.com/photoview/photoview/pull/940
* Refactor docker deployment and user setup by @kkovaletp in https://github.com/photoview/photoview/pull/863
* Allow SQLite to accept several connections at the same time by @kkovaletp in https://github.com/photoview/photoview/pull/955

### Fixed
* Fix transaction already committed error by @Ajnasz in https://github.com/photoview/photoview/pull/795
* Fix translation error of Chinese UI in a new branch by @WindLi001 in https://github.com/photoview/photoview/pull/796
* Use Clean() but not Abs() to prevent directory traversal by @WindLi001 in https://github.com/photoview/photoview/pull/799
* Fix the "HEIC: failed to parse Exif data: exiftool buffer too small" issue by @tymmej in https://github.com/photoview/photoview/pull/807
* Fix encoding video thumbnails by @tymmej in https://github.com/photoview/photoview/pull/810
* Prepend missing key CreationDate to createDateKeys of FFmpeg parser by @yosev in https://github.com/photoview/photoview/pull/822
* Fix custom basepath in React components by @djvaldez in https://github.com/photoview/photoview/pull/838
* Fix build pipeline by @viktorstrate in https://github.com/photoview/photoview/pull/865
* Fix UI not serving media and tests expecting localhost:3000 as base URL by @davidecavestro in https://github.com/photoview/photoview/pull/874
* Fix invalid UTC offset during date-time parsing by @yosev in https://github.com/photoview/photoview/pull/823
* Fix audit issues in production by @jordy2254 in https://github.com/photoview/photoview/pull/933
* Fix false trigger of DB migration on SQLite by @kabakaev in https://github.com/photoview/photoview/pull/936
* Fix download path on custom base path by @djvaldez in https://github.com/photoview/photoview/pull/889
* Dockerfile: Fail chained command in RUN if some of the commands fail by @kkovaletp in https://github.com/photoview/photoview/pull/947
* Fix timeline order with SQLite3. It should be ordered as MySQL and PGSQL like `(year, month, day)`. by @googollee in https://github.com/photoview/photoview/pull/954
* Update the `healthcheck` command for the MariaDB container by @kkovaletp in https://github.com/photoview/photoview/pull/963
* Fix typo in the `serviceWorkerRegistration.ts` by @kkovaletp in https://github.com/photoview/photoview/pull/964
* Freezing UID and GID of the `photoview` user in the Dockerfile by @kkovaletp in https://github.com/photoview/photoview/pull/972

## New Contributors
* @eadisor made their first contribution in https://github.com/photoview/photoview/pull/651
* @rainpenber made their first contribution in https://github.com/photoview/photoview/pull/749
* @noel123007 made their first contribution in https://github.com/photoview/photoview/pull/774
* @queeup made their first contribution in https://github.com/photoview/photoview/pull/780
* @Ajnasz made their first contribution in https://github.com/photoview/photoview/pull/795
* @WindLi001 made their first contribution in https://github.com/photoview/photoview/pull/796
* @yosev made their first contribution in https://github.com/photoview/photoview/pull/802
* @tymmej made their first contribution in https://github.com/photoview/photoview/pull/807
* @djvaldez made their first contribution in https://github.com/photoview/photoview/pull/838
* @secco04 made their first contribution in https://github.com/photoview/photoview/pull/841
* @davidecavestro made their first contribution in https://github.com/photoview/photoview/pull/873
* @kkovaletp made their first contribution in https://github.com/photoview/photoview/pull/875
* @AndGem made their first contribution in https://github.com/photoview/photoview/pull/917
* @cicadabear made their first contribution in https://github.com/photoview/photoview/pull/888
* @DorianMazur made their first contribution in https://github.com/photoview/photoview/pull/909
* @alcaprar made their first contribution in https://github.com/photoview/photoview/pull/851
* @kabakaev made their first contribution in https://github.com/photoview/photoview/pull/883
* @subtlepseudonym made their first contribution in https://github.com/photoview/photoview/pull/828
* @emersion made their first contribution in https://github.com/photoview/photoview/pull/881
* @jordy2254 made their first contribution in https://github.com/photoview/photoview/pull/933
* @googollee made their first contribution in https://github.com/photoview/photoview/pull/954

**Full Changelog**: https://github.com/photoview/photoview/compare/v2.3.13...v2.4.0

<hr /><em>This discussion was created from the release <a href='https://github.com/photoview/photoview/releases/tag/v2.4.0'>Project restart: a lot of improvements!</a>.</em>

**Comment by hukl** [comment]
Oh I thought this was abandon ware - super excited to see it's still going. Just tested my build procedure for FreeBSD and it still works nicely. Thanks for the effort! One question though: Is Viktor still actively involved with this project and if not, is the github sponsor money distributed to the active developers?

**Reply by kkovaletp** [reply]
> One question though: Is Viktor still actively involved with this project and if not, is the github sponsor money distributed to the active developers?

Let's tag @viktorstrate, so that he can see and answer this question from his point of view)

I could say that he is still involved, but more like a consultant, while the active product maintenance is done by me; PR review and some help from the development side are done by @jordy2254.

I didn't get any money from the project (not as a maintainer, not as a contributor) and I assume that @jordy2254 didn't as well (he can confirm or correct me if I'm wrong).

**Reply by hukl** [reply]
Ok - I'd happily continue to support the project with a bit of € but I'd love to see the people doing the bulk of the work receiving it. Maybe you can discuss and decide how you want to handle it.

**Reply by kkovaletp** [reply]
Thanks a lot for your support!

Right now the project needs more experienced developers (frontend and backend), willing to help with the code reviews of existing PRs, answer questions about some development issues and corner cases, and, of course, contribute new PRs with fixes and improvements for the reported issues. Currently, this is the bottleneck of the project's development speed.

If you are one of them or know someone, interested in taking this responsibility (and committing to invest at least some minimal time regularly), doing these activities for the project, we all in the Photoview Community would really appreciate and welcome to the maintainers' group)

**Comment by greenship24** [comment]
Just for some people with K8s that might need to make changes for this release- this is what I did.
- changed PHOTOVIEW_LISTEN_PORT to 8080 so the non-root user doesn't have to bind to a privileged port
- Adjusted my service to still listen on port 80 but targetPort is 8080 (so didn't have to mod my ingress)
- `chown -R 999 /path/to/cache` to reflect new non-root user ID 
- Otherwise you can keep all capabilities dropped and allowPrivilegeEscalation set to false (yay security)

Thanks to all the contributors. I've been playing around with this in a test instance and it seems to work fantastically and fixes some small bugs I had on the previous version.

**Comment by anyshpm** [comment]
Great works.
